crates/core/src/prelude/src/memory.ts

```diff
@@ -0,0 +1,86 @@
+declare global {
+  interface Memory {
+    fromString(str: string): MemoryHandle;
+    fromBuffer(bytes: ArrayBufferLike): MemoryHandle;
+    fromJsonObject(obj: JSON): MemoryHandle;
+    allocUInt32(i: number): MemoryHandle;
+    allocUInt64(i: bigint): MemoryHandle;
+    allocFloat32(i: number): MemoryHandle;
+    allocFloat64(i: number): MemoryHandle;
+    find(offset: number): MemoryHandle;
+
+    /**
+     * @internal
+     */
+    _fromBuffer(buffer: ArrayBuffer): { offset: number; len: number };
+    /**
+     * @internal
+     */
+    _find(offset: number): { offset: number; len: number };
+    /**
+     * @internal
+     */
+    _free(offset: number): void;
+    /**
+     * @internal
+     */
+    _readBytes(offset: number): ArrayBuffer;
+  }
+  var Memory: Memory;
+}
+
+const Memory = globalThis.Memory;
+Memory.fromString = function (this: Memory, str) {
+  // todo validate
+  let bytes = new TextEncoder().encode(str).buffer;
+  const memData = Memory.fromBuffer(bytes);
+  return new MemoryHandle(memData.offset, memData.len);
+};
+
+Memory.fromBuffer = function (this: Memory, bytes) {
+  // todo validate
+  const memData = Memory._fromBuffer(bytes);
+  return new MemoryHandle(memData.offset, memData.len);
+};
+
+Memory.fromJsonObject = function (this: Memory, obj) {
+  // todo validate
+  const memData = Memory.fromString(JSON.stringify(obj));
+  return new MemoryHandle(memData.offset, memData.len);
+};
+
+Memory.allocUInt32 = function (this: Memory, i) {
+  const buffer = new ArrayBuffer(4);
+  const arr = new Uint32Array(buffer);
+  arr[0] = i;
+  return Memory.fromBuffer(buffer);
+};
+
+Memory.allocUInt64 = function (this: Memory, i) {
+  const buffer = new ArrayBuffer(8);
+  const arr = new BigUint64Array(buffer);
+  arr[0] = i;
+  return Memory.fromBuffer(buffer);
+};
+
+Memory.allocFloat32 = function (this: Memory, i) {
+  const buffer = new ArrayBuffer(4);
+  const arr = new Float32Array(buffer);
+  arr[0] = i;
+  return Memory.fromBuffer(buffer);
+};
+
+Memory.allocFloat64 = function (this: Memory, i) {
+  const buffer = new ArrayBuffer(8);
+  const arr = new Float64Array(buffer);
+  arr[0] = i;
+  return Memory.fromBuffer(buffer);
+};
+
+Memory.find = function (offset) {
+  // todo validate
```

Comment:
Let's make sure we get a resolution [here](https://github.com/extism/js-pdk/pull/60) as the convo has moved.

Reply:
@bhelx, do you recall leaving that comment? It wasn't mine, I just copied it over when moving some of this code, so unsure what would need to be done.

Reply:
hmm no i don't. If I did then i'm not sure that to validate. I assumed maybe @zshipko added it. I think because it's typescript we should be okay. the Extism memory should break if the offset is not a valid offset otherwise.

Reply:
i don't remember adding that either, but it looks like it's referring to what @chrisdickinson was talking about. do we want a failed lookup to return `{offset: 0, length: 0}`? we could also return `null` or raise an exception?